model/utils/mlflow.py

```diff
@@ -17,29 +15,27 @@ def close(self):
         mlflow.end_run()
 
     def login_to_mlflow(
-        self, uri: str = "databricks", experiment_name: str = "/Users/michal.zareba@softwaremill.pl/housepricesml"
+        self, uri: str = "http://48.209.80.111:5000", experiment_name: str = "housepricesml"
     ) -> None:
-        mlflow.login()
         mlflow.set_tracking_uri(uri)
         mlflow.set_experiment(experiment_name)
-        mlflow.set_registry_uri("databricks-uc")
 
-    def log_preprocessed_data(self, df: pd.DataFrame) -> None:
-        df.to_csv("preprocessed_data.csv")
-        mlflow.log_artifact("preprocessed_data.csv")
-        os.remove("preprocessed_data.csv")
+    def log_preprocessed_data(self, df: pd.DataFrame, path: Path = Path("preprocessed_data.csv")) -> None:
+        df.to_csv(path)
+        mlflow.log_artifact(path)
+        os.remove(path)
```

Comment:
Here you are providing a path to the file that you are creating and then deleting. This is the perfect case for using temporary files. 

```python
import tempfile

with tempfile.NamedTemporaryFile() as file:
    ...
```